pkg/mount/dag_steps.go

```diff
@@ -384,10 +385,30 @@ func (s *State) WriteSentinelDagStep(g *herd.Graph, deps ...string) error {
 			// Lets add a uki sentinel as well!
 			cmdline, _ := os.ReadFile(internalUtils.GetHostProcCmdline())
 			if strings.Contains(string(cmdline), "rd.immucore.uki") {
+				// Generic sentinel for uki mode
+				// TODO: Drop this from the layout file and use the ones below to check our uki mode
 				err = os.WriteFile("/run/cos/uki_mode", []byte("1"), os.ModePerm)
+				internalUtils.Log.Info().Str("to", "uki_mode").Msg("Setting sentinel file")
 				if err != nil {
 					return err
 				}
+				// Specific one
+				err = internalUtils.CheckEfiPartUUID()
+				if err != nil {
+					internalUtils.Log.Info().Str("to", "uki_install_mode").Msg("Setting sentinel file")
+					err2 := os.WriteFile("/run/cos/uki_install_mode", []byte("1"), os.ModePerm)
+
+					if err2 != nil {
+						return err2
+					}
+					return nil
```

Comment:
```suggestion
					return os.WriteFile("/run/cos/uki_install_mode", []byte("1"), os.ModePerm)
```

Reply:
not really because I want to return an error only if writing the file failed, otherwise it means that we are not on install mode and we cannot return an error.

I think...

Reply:
but if no error occurrs then you get nil from WriteFile, no? so you cannot distinguish from these scenarios

Reply:
yes, basically if we cannot read the efivar file or its empty, we get an error from internalUtils.CheckEfiPartUUID() but we dont want to return that error, that its just an internal way of checking.


Maybe the problem here is that Im using an error to identify this.... and it should be a boolean